scripts/lazy-load-images.js

```diff
@@ -1,20 +1,56 @@
-const inView = require('in-view')
+import { throttle } from 'lodash'
+
+function inViewport (element) {
+  const { top, right, bottom, left } = element.getBoundingClientRect()
+  return bottom > 0
+      && window.innerWidth - left > 0
+      && window.innerHeight - top > 0
+      && right > 0
+}
 
 module.exports = function () {
-  inView('img[data-src]')
-    .on('enter', img => {
-      // abort if this element has previously entered the viewport
-      if (!img.dataset.src) return
-
-      img.setAttribute('src', img.dataset.src)
-
-      // remove attribute so CSS fade-in can happen
-      img.onload = () => {
-        img.removeAttribute('data-src')
-        
-        // remove colored circular placeholder element
-        const placeholder = img.parentElement.querySelector('.placeholder')
-        if (placeholder) placeholder.classList.add('faded-out')
-      }
-    })
-}
\ No newline at end of file
+  ['scroll', 'resize', 'load'].forEach(event =>
+    window.addEventListener(event, updateImages)
+  )
+
+  // add all images that are part of the DOM tree
+  // NB: |display: none| elements need manual adding
+  document.querySelectorAll('img[data-src]').forEach(addImage)
+}
+
+let images = []
+
+let addImage = module.exports.addImage = function (image) {
+  // abort if this element has previously entered the viewport
+  if (!image.dataset.src) return
+
+  images.push(image)
+  updateImages()
+}
+
+function handle (image) {
```

Comment:
Can you use a more descriptive name for this method, like `displayImage` or similar?